### PR TITLE
fix(pty): remove port validation from WebSocket URL check

### LINKED ISSUE #226: Overly strict PTY WebSocket URL validation rejects legitimate Websh server connections
## Description

`validateWebSocketURL` in `pkg/runner/pty.go` compares the PTY WebSocket URL's hostname and port against the `server.url` in `alpamon.conf`, rejecting connections when they don't match exactly.

The Websh (PTY WS) proxy server runs on the **same hostname** as the Alpacon API server but on a **different port** in local development. This causes PTY session initialization to always fail.

## Steps to Reproduce

1. Configure `alpamon.conf` with `url = http://localhost:8000`
2. Run alpamon and connect to the Alpacon console
3. Open a PTY terminal session from the console

## Expected Behavior

PTY WebSocket connection should be established successfully, as the Websh proxy server is a legitimate service on the same host.

## Actual Behavior

PTY session initialization fails immediately with:
```
[ERROR] Failed to initialize PTY session. error="WebSocket URL port \"8080\" does not match server port \"8000\""
```

## Cause

Port validation was added in commit `e0584fc` (2026-03-07) as part of Copilot review hardening on top of the original CodeQL SSRF fix.

```go
// pkg/runner/pty.go - validateWebSocketURL()

// Port validation — fails when ports differ; guesses defaults (443/80) when not specified
parsedPort := normalizePort(parsed.Scheme, parsed.Port())
serverPort := normalizePort(serverURL.Scheme, serverURL.Port())
if parsedPort != "" && serverPort != "" && parsedPort != serverPort {
    return fmt.Errorf(...)
}
```

## Infrastructure

| Environment | Alpacon API | Websh (PTY WS) |
|-------------|-------------|-----------------|
| Production | `https://workspace.region.alpacon.io` | `wss://workspace.region.alpacon.io/ws/websh/...` (same host, different path) |
| Local dev | `http://localhost:8000` | `ws://localhost:8080` (same host, different port) |

- The PTY WebSocket URL is provided by the authenticated Alpacon console as part of the `openpty` command
- Production: same hostname, routed via LB/ingress to the proxy server
- Local dev: same hostname (localhost), different port (API=8000, proxy=8080)

## Proposed Fix

- **Keep scheme validation**: Derive expected WS scheme from server URL (`http` → `ws`, `https` → `wss`)
- **Keep hostname validation**: Exact hostname match (same host in both production and local dev)
- **Remove port validation**: Port differs in local dev; guessing default ports (443/80) is unreliable
- **Remove `normalizePort` helper**: No longer needed

### PR DESCRIPTION
**Pull request by jisung-02**
## Summary

Remove port comparison and `normalizePort` helper from `validateWebSocketURL`, keeping only scheme and hostname validation. The proxy server runs on a different port (8080) than the API server (8000) in local dev, causing PTY session initialization to always fail. Port validation was not part of the original CodeQL SSRF fix — it was added later as over-hardening in `e0584fc`.

## Changes

- Remove port comparison logic from `validateWebSocketURL` in `pkg/runner/pty.go`
- Remove `normalizePort` helper function
- Remove port-related test cases (`TestValidateWebSocketURL_ServerWithExplicitPort`, `TestNormalizePort`)

## Context

PR #221 added `validateWebSocketURL` to address a CodeQL SSRF alert. The original fix (scheme + hostname check) was correct, but a subsequent Copilot review added port comparison with a `normalizePort` helper that guesses default ports (443/80).

This breaks PTY connections because:
- **Local dev**: API server on `:8000`, proxy server on `:8080` → port mismatch → rejected
- **Production**: Same host behind LB, but guessing default ports is unreliable when ports are implicit in URLs

Other WebSocket connections in the codebase (FTP, tunnel, main client, control client) do not validate URLs at all — tunnel explicitly comments "Server URL is provided by the authenticated Alpacon console which the agent trusts."

## Security

Scheme + hostname validation is sufficient for SSRF protection:
- **Scheme validation**: Only `ws`/`wss` allowed, blocking SSRF via `http`, `ftp`, etc.
- **Hostname validation**: Ensures the URL matches the configured server host (e.g., `alpacax.dev.alpacon.io`), blocking requests to external hosts like `evil.com`
- Port is not a CodeQL SSRF check criterion — different ports on the same trusted host are not a security threat

Closes #226

### Universal
- [x] Code follows project style guidelines
- [x] Self-reviewed the code
- [x] No hardcoded secrets or credentials

### Golang
- [x] Error handling is complete
- [x] Tests include edge cases
- [x] go mod tidy executed

### CODE REVIEW

File: pkg/runner/pty.go [hunk 1]
```diff
@@ -433,30 +433,9 @@ func validateWebSocketURL(rawURL string) error {
 		return fmt.Errorf("WebSocket URL host %q does not match server host %q", parsed.Hostname(), serverURL.Hostname())
 	}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply pkg/runner/pty.go:434]: The issue description was inaccurate and has been updated — the Websh proxy actually runs on the same hostname in both production (wss://workspace.region.alpacon.io/ws/websh/...) and local dev (ws://localhost:8080), so exact hostname match is correct.

File: pkg/runner/pty.go [hunk 2]
```diff
@@ -433,30 +433,9 @@ func validateWebSocketURL(rawURL string) error {
...
 
-	parsedPort := normalizePort(parsed.Scheme, parsed.Port())
-	serverPort := normalizePort(serverURL.Scheme, serverURL.Port())
-	if parsedPort != "" && serverPort != "" && parsedPort != serverPort {
-		return fmt.Errorf("WebSocket URL port %q does not match server port %q", parsedPort, serverPort)
-	}
-
 	return nil
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply pkg/runner/pty.go:436]: Good point — added TestValidateWebSocketURL_ServerWithExplicitPort to cover this scenario (same port, different port, no port all allowed; different host rejected).

**Review by eunyoung14 — APPROVED** [review]
